flags_test.go

```diff
@@ -357,3 +361,117 @@ func (t *FlagsTest) TestValidateFlagsForValidSequentialReadSizeAndHTTP2ClientPro
 
 	AssertEq(nil, err)
 }
+
+func (t *FlagsTest) TestMetadataCacheTTL() {
+	inputs := []struct {
+		// equivalent of user-setting of --stat-cache-ttl
+		statCacheTTL time.Duration
+
+		// equivalent of user-setting of --type-cache-ttl
+		typeCacheTTL time.Duration
+
+		// equivalent of user-setting of metadata-cache:ttl-secs in --config-file
+		ttlInSeconds             int64
+		expectedMetadataCacheTTL time.Duration
+	}{
+		{
+			// most common scenario, when user doesn't set any of the TTL config parameters.
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             config.TtlInSecsUnsetSentinel,
+			expectedMetadataCacheTTL: DefaultStatOrTypeCacheTTL,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to -1
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             -1,
+			expectedMetadataCacheTTL: time.Duration(math.MaxInt64),
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to 0
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             0,
+			expectedMetadataCacheTTL: 0,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to a positive value
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             30,
+			expectedMetadataCacheTTL: 30 * time.Second,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to its highest supported value
+			statCacheTTL: DefaultStatOrTypeCacheTTL,
+			typeCacheTTL: DefaultStatOrTypeCacheTTL,
+			ttlInSeconds: config.MaxSupportedTtlInSeconds,
+
+			// math.MaxInt64 is not a multiple of 1e9, so time.Duration(math.MaxInt64) is not
+			// an exact multiple of seconds. Since ttlInSeconds will always set time in
+			// multiple of seconds, we also need to round time.Duration(math.MaxInt64) to
+			// multiple of seconds, for comparison.
+			expectedMetadataCacheTTL: time.Duration((math.MaxInt64 / int64(time.Second)) * int64(time.Second)),
```

Comment:
why are we not using config.MaxSupportedTtlInSeconds directly here ?

Reply:
It's explained in the comment above already. Basically, config.MaxSupportedTtlInSeconds i.e. time.duration(MaxInt64) doesn't represent an actual multiple of a second.
We could either redefine MaxSupportedTtlInSeconds itself to be the truncated value, or keep this test this way.

Reply:
Will change it to `time.Second * time.Duration(config.MaxSupportedTtlInSeconds)` .